block-producer/src/batch_builder/mod.rs

```diff
@@ -111,17 +112,19 @@ impl<BB> BatchBuilder for DefaultBatchBuilder<BB>
 where
     BB: BlockBuilder,
 {
+    #[instrument(target = "miden-block-producer", skip_all)]
     async fn build_batch(
         &self,
         txs: Vec<SharedProvenTx>,
     ) -> Result<(), BuildBatchError> {
         let num_txs = txs.len();
 
+        info!(target: COMPONENT, num_txs, "Batch building started");
+        debug!(target: COMPONENT, txs = %format_array(txs.iter().map(|tx| tx.id().to_hex())));
+
         let batch = Arc::new(TransactionBatch::new(txs)?);
         self.ready_batches.write().await.push(batch);
 
-        info!(COMPONENT, "batch built with {num_txs} txs");
-
```

Comment:
I would not remove this but modify to log "Transaction batch built". We should log the batch ID (see next comment) and the total number of batches now in the queue (similar to what we did in the submit proven transaction flow).

For login batch ID, we could also do it as a part of `instrument` on line 115. For this, we could declare a field there and provide the value of this field later (see [here](https://docs.rs/tracing/latest/tracing/attr.instrument.html) under "Adding an empty field to be recorded later").